internal/handler/wallet/get.go

```diff
@@ -26,6 +40,30 @@ func HandleGet(params operations.RestWalletGetParams) middleware.Responder {
 
 	modelWallet := createModelWallet(*wlt)
 
+	// if request not ciphered data, ask for password and unprotect the wallet
+	if params.Ciphered != nil && !*params.Ciphered {
+		promptData := &wallet.PromptRequestData{
+			Msg:  fmt.Sprintf("Unprotect wallet %s", wlt.Nickname),
+			Data: nil,
+		}
+		_, err := wlt.PromptPassword(g.prompterApp, walletapp.Export, promptData)
+		if err != nil {
+			return operations.NewRestWalletGetLocked().WithPayload(
+				&models.Error{
+					Code:    errorGetWallets,
+					Message: "Unable to unprotect wallet",
+				})
+		}
+		salt := base58.CheckEncode(wlt.KeyPair.Salt[:], wallet.Base58Version)
+		nonce := base58.CheckEncode(wlt.KeyPair.Nonce[:], wallet.Base58Version)
+		modelWallet.KeyPair = models.WalletKeyPair{
```

Comment:
Sucess event must be emited

Reply:
why success event is not emitted in func (w *Wallet) PromptPassword ?

Reply:
je l'ai pas mis dedans car sa permet de faire d'autres operation avant de l'envoyer. par exemple pour le delete je supprime le fichier avant